### PR TITLE
Make UIP_CONF_DS6_ROUTE_NBU and DS6_NBR_NBU, DS6_ROUTE_NBU obsolete

### PR DESCRIPTION
**Pull request by MohamedSeliem**
update uip-ds.h comments, this small change is made to ease the track of Tables implementation places

remove definition of UIP_CONF_DS6_ROUTE_NBU

retire 2 old macros DS6_NBR_NBU, DS6_ROUTE_NBU

### CODE REVIEW

**Comment by MohamedSeliem** [comment]
@simonduq  i combined the modifications in #1770 #1771 , and squashed them in one commit based on your comments.
"removing all definitions of UIP_CONF_DS6_NBR_NBU and UIP_CONF_DS6_ROUTE_NBU (not only in the multicast example)" :)

File: core/net/ipv6/uip-ds6.h [hunk 1]
```diff
@@ -51,11 +51,14 @@
...
- * - the number of elements assigned by the system (name suffixed by _NBS)
- * - the total number of elements is the sum (name suffixed by _NB)
+* Default Router List, Unicast address list, multicast address list, anycast address list),
+* we define:
+* - the number of elements requested by the user in contiki configuration (name suffixed by _NBU)
+* - the number of elements assigned by the system (name suffixed by _NBS)
+* - the total number of elements is the sum (name suffixed by _NB)
+* 
+* The routing table definitions can be found in uip-ds6-route.h
+* The Neighbor cache definitions can be found in nbr-table.h
```
Comment (simonduq) [inline core/net/ipv6/uip-ds6.h]: there seems to be an unintended change of indentation here

**Comment by simonduq** [comment]
Looks good to me except for some unwanted indentation change in comments

**Comment by MohamedSeliem** [comment]
already fixed ;)

**Comment by simonduq** [comment]
did you push? the diff still shows a change in indent

**Comment by MohamedSeliem** [comment]
done

**Comment by simonduq** [comment]
No sorry, what I mean is the space you removed before the `*` in `uip-ds6.h`, lines 54 to 60.
Please rebase after that to avoid unnecessary commits (just use "fixup"). Then just push -f

**Comment by MohamedSeliem** [comment]
Thank you.

**Comment by simonduq** [comment]
Great. Thanks. :+1: if Travis is green

**Comment by MohamedSeliem** [comment]
@simonduq so any plans to merge these changes soon?

**Comment by simonduq** [comment]
see https://github.com/contiki-os/contiki/blob/master/CONTRIBUTING.md#pull-request-merging-policy

**Comment by MohamedSeliem** [comment]
okay, Thanks.

**Comment by nvt** [comment]
:+1:

**Pull request merged**